documentation/add/04_mpi_framework.md

```diff
@@ -1,43 +1,244 @@
+
+#### Note^2 Persistent messages
+Persistent messages are used to distribute information about special states of the system, i.e. failure or job completed states as parallel interrupts, used for these purposes in standard MPI frameworks are not available for all parallel frameworks. 
+
+#### Note^3 Messages types
+
+Different purposes can be best served using different types of messages. There are 3 purposes, differentiated in Horace.
+
+    1) logging, diagnostics and informing user about progress of a job. 
+    2) Interprocess data exchange, necessary for performing a particular user job and 
+    3) The information about error, failure or job completion. 
+    
+Different types of messages created to serve correspondent purposes:
+
+    1) Non-blocking (transient)  messages. Informing user about a job progress is important task, but user is interested only in the final value. Different independent task should not be synchronous with regard to log messages as only some average progress is required. Non-blocking messages are best for such kind of task.
+    2) Blocking messages are necessary to transfer job information between independent workers when the results of calculations on a single node depends on the results on other nodes.
+    3) Persistent messages necessary to indicate some critical conditions or error states, as i.e, parallel interrupts, available in MPI for such purposes are not available for file-based system (*ClusterHerbert*) and not accessible for Matlab MPI (*ClusterParpool*).
+
+To provide the described flexibility and variability of messages, all messages used by Horace MPI are children of **aMessage**  class and subscribed to a messages factory. 
+
+## Messages class and messages factory.
+
+The ordinary message can be defined as member of **aMessage** class with a specific name. Some messages need additional functionality so additional properties are defined for children classes, describing these messages. Current family of the message classes is defined on the **Fig 6**
+
+![Fig 6: Messages Family](../diagrams/aMessagesTree.png)
```

Comment:
Do the `loadobj` / `saveobj` methods allow for the messages to be encoded in anything other than Matlab binary format? 

How do these four concrete `aMessage` instances map onto the three message types discussed above?

Are there any restrictions/limitations on the `payload`?

Reply:
I hope I have replied this question by modyfying the class and introducing appropriate properties.